pkg/network/usm/sharedlibraries/ebpf.go

```diff
@@ -126,19 +181,182 @@ func (e *EbpfProgram) Init() error {
 		log.Warnf("runtime compilation failed: attempting fallback: %s", err)
 	}
 
-	return e.initPrebuilt()
+	if err := e.initPrebuilt(); err != nil {
+		return fmt.Errorf("prebuilt load failed: %w", err)
+	}
+
+	return nil
 }
 
-// GetPerfHandler returns the perf handler
-func (e *EbpfProgram) GetPerfHandler() *ddebpf.PerfHandler {
-	return e.perfHandler
+// InitWithLibsets initializes the eBPF program. It is guaranteed to perform the initialization only if needed
+// For example, if the program is already initialized to listen for a certain libset, it will not reinitialize
+// the program to listen for the same libset. However, if the libsets are changed, it will reinitialize the program.
+func (e *EbpfProgram) InitWithLibsets(libsets ...Libset) error {
+	// Ensure we have all valid libsets, we don't want cryptic errors later
+	for _, libset := range libsets {
+		if !IsLibsetValid(libset) {
+			return fmt.Errorf("libset %s is not valid, ensure it is in the LibsetToLibSuffixes map", libset)
+		}
+	}
+
+	// Lock for the initialization variables
+	e.initMutex.Lock()
+	defer e.initMutex.Unlock()
+
+	// If the program is initialized, check if the libsets are already enabled
+	if e.isInitialized {
+		// If the libsets are already enabled, return, we have nothing to do
+		if e.areLibsetsAlreadyEnabled(libsets...) {
+			return nil
+		}
+
+		// If not, we need to reinitialize the eBPF program, so we stop it
+		// We use stopImpl to avoid changing the refcount
+		e.stopImpl()
```

Comment:
why reinitialize?

Reply:
The constants are edited in `InitWithOptions`, and we cannot call that multiple times unless we cleanup the manager and recreate it. I added a comment that need.